apps/server/src/modules/learnroom/service/course-sync.service.ts

```diff
@@ -42,36 +51,44 @@ export class CourseSyncService {
 	}
 
 	private async synchronize(courses: Course[], group: Group, oldGroup?: Group): Promise<void> {
-		if (courses.length) {
-			const [studentRole, teacherRole] = await Promise.all([
-				this.roleService.findByName(RoleName.STUDENT),
-				this.roleService.findByName(RoleName.TEACHER),
-			]);
-			const students = group.users.filter((groupUser: GroupUser) => groupUser.roleId === studentRole.id);
-			const teachers = group.users.filter((groupUser: GroupUser) => groupUser.roleId === teacherRole.id);
-
-			const coursesToSync = courses.map((course) => {
-				course.syncedWithGroup = group.id;
-				if (oldGroup && oldGroup.name === course.name) {
-					course.name = group.name;
-				}
-				course.startDate = group.validPeriod?.from;
-				course.untilDate = group.validPeriod?.until;
-
-				if (teachers.length >= 1) {
-					course.students = students.map((user: GroupUser): EntityId => user.userId);
-					course.teachers = teachers.map((user: GroupUser): EntityId => user.userId);
-				} else {
-					course.students = [];
-				}
-
-				course.classes = [];
-				course.groups = [];
-
-				return course;
-			});
-
-			await this.courseRepo.saveAll(coursesToSync);
+		const [studentRole, teacherRole] = await Promise.all([
+			this.roleService.findByName(RoleName.STUDENT),
+			this.roleService.findByName(RoleName.TEACHER),
+		]);
+
+		const studentIds = group.users
+			.filter((user: GroupUser) => user.roleId === studentRole.id)
+			.map((student) => student.userId);
+		const teacherIds = group.users
+			.filter((user: GroupUser) => user.roleId === teacherRole.id)
+			.map((teacher) => teacher.userId);
+
+		for (const course of courses) {
+			course.syncedWithGroup = group.id;
+			course.startDate = group.validPeriod?.from;
+			course.untilDate = group.validPeriod?.until;
+			course.classes = [];
+			course.groups = [];
+
+			if (oldGroup?.name === course.name) {
+				course.name = group.name;
+			}
+
+			const excludedFromSync = new Set(course.excludeFromSync || []);
+
+			// always sync teachers unless explicitly excluded
```

Comment:
in this project comments are "discouraged" 😅 The mantra here is "code is doc". Only if it is something one would not understand without the comment. I think we can forego this one. If we need the next one is a good question, because without the knowledge of the explicit requirement it doesn't make any sense ^^ We should talk about this tomorrow / in a few hours :)